lxd/container_post.go

```diff
@@ -203,12 +244,153 @@ func containerPost(d *Daemon, r *http.Request) Response {
 	return OperationResponse(op)
 }
 
-// Special case migrating a container backed by ceph across two cluster nodes.
-func containerPostClusteringMigrateWithCeph(d *Daemon, c container, oldName, newName, newNode string) Response {
-	if c != nil && c.IsRunning() {
-		return BadRequest(fmt.Errorf("Container is running"))
+// Move a non-ceph container to another cluster node.
+func containerPostClusteringMigrate(d *Daemon, c container, oldName, newName, newNode string) Response {
+	cert := d.endpoints.NetworkCert()
+
+	var sourceAddress string
+	var targetAddress string
+
+	// Save the original value of the "volatile.apply_template" config key,
+	// since we'll want to preserve it in the copied container.
+	origVolatileApplyTemplate := c.LocalConfig()["volatile.apply_template"]
+
+	err := d.cluster.Transaction(func(tx *db.ClusterTx) error {
+		var err error
+
+		sourceAddress, err = tx.NodeAddress()
+		if err != nil {
+			return errors.Wrap(err, "Failed to get local node address")
+		}
+
+		node, err := tx.NodeByName(newNode)
+		if err != nil {
+			return errors.Wrap(err, "Failed to get new node address")
+		}
+		targetAddress = node.Address
+
+		return nil
+	})
+	if err != nil {
+		return SmartError(err)
 	}
 
+	run := func(*operation) error {
+		// Connect to the source host, i.e. ourselves (the node the container is running on).
+		source, err := cluster.Connect(sourceAddress, cert, false)
+		if err != nil {
+			return errors.Wrap(err, "Failed to connect to local node")
+		}
+
+		// Connect to the destination host, i.e. the node to migrate the container to.
+		dest, err := cluster.Connect(targetAddress, cert, false)
+		if err != nil {
+			return errors.Wrap(err, "Failed to connect to local node")
+		}
+		dest = dest.UseTarget(newNode)
+
+		destName := newName
+		isSameName := false
+
+		// If no new name was provided, the user wants to keep the same
+		// container name. In that case we need to generate a temporary
+		// name.
+		if destName == "" || destName == oldName {
+			isSameName = true
+			destName = fmt.Sprintf("move-%s", uuid.NewRandom().String())
+		}
+
+		// First make a copy on the new node of the container to be moved.
+		entry, _, err := source.GetContainer(oldName)
+		if err != nil {
+			return errors.Wrap(err, "Failed to get container info")
+		}
+
+		args := lxd.ContainerCopyArgs{
+			Name:          destName,
+			Live:          true,
```

Comment:
Hmm, Live should be set to False here (just not set) as we're not doing live migration.

Reply:
Fixed.